{{ cookiecutter.repo_name }}/data/README.md

```diff
@@ -0,0 +1,25 @@
+# `data` folder
+
+Any data that needs to be stored locally should be saved in this location. This folder, and all its sub-folders are not
+version-controlled.
+
+The sub-folders should be used as follows:
+
+- `external`: Any data that will **not** be processed at all, such as reference data;
+- `raw`: Any raw data before any processing;
+- `interim`: Any raw data that has been partially processed and, for whatever reason, needs to be stored before further 
+processing is completed; and
+- `processed`: Any raw or interim data that has been fully processed into its final state.
```

Comment:
If any file is already tracked **before** `.gitignore` is used to skip the entire folder, then it will remain in tracked in the Git history - `.gitkeep` is just a convenience name for a tracked file in an untracked folder, and has no actual effect.

Note although the above approach works in 'standard' repositories that already have Git set up, it has caused a bug here with the cookiecutter output. The first user will see these `data` folder and sub-folders, but these will not be pushed to the remote.

This is because Git isn't set up in the cookiecutter output, so no files are tracked when the cookiecutter output is first generated, i.e. these `.gitkeep` files are not tracked. When version control is established, Git will automatically ignore all these folders **including the `.gitkeep` files** due to the `.gitignore`. The user would have to manually track these files, which is of course not ideal.

Adding a `.gitignore` in each sub-folder would fix this, although it would be more effort if you wanted to make global changes, as you'd have to go into each `.gitignore` and make the same change (unlikely in our case, but still a possibility).

Will implement a fix that solves these issues based on this [StackOverflow answer](https://stackoverflow.com/a/20652768) - this implements your multiple `.gitignore` approach @exfalsoquodlibet but ensures all parameters are in a single `.gitignore`.

Reply:
Made the fix in commit [`c1c8383`](https://github.com/ukgovdatascience/govcookiecutter/pull/1/commits/c1c8383877b486d716e165573367b5526c1ed083).

Reply:
Interesting solution - I'll copy this :)